pkgs/pkgs-lib/formats.nix

```diff
@@ -120,18 +122,26 @@ rec {
         else
           singleIniAtom;
 
-    in attrsOf (attrsOf iniAtom);
+    in
+      if withGlobalSection then attrsOf (attrsOf (either iniAtom (attrsOf iniAtom)))
```

Comment:
Adding type hints that this attrset has exactly two keys (`globalSection` and `sections`) rather than accepting any attrset would be nice in case anyone typo'd or something. The code specifically only passes those options on to the generator function, which'd mean passing something like `{ foo = "bar"; globalSection = {}; sections = {}; }` might go unnoticed.

IMHO not a strict requirement, but an improvement.

Reply:
Thanks for pointing this out. This is actually a strict requirement. In particular, it should hold that if `format.type` is valid for a value, `format.generate` should not throw an error for that value.

And doing this will indeed break the example module you gave in the description @tim-tx